### PR TITLE
Feature tests

### PR DESCRIPTION
**Pull request by jackha**
Added a bunch of unit tests:
- Settings
- AABB
- AABB3D

### CODE REVIEW

File: tests/utils/AABB3DTest.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    void includeZTest();
+    void offsetTest();
+    void offsetTest2();
+    void expandXYTest();
+
+private:
+    /*!
+     * \brief The maximum allowed error in floats.
+     */
+    static constexpr double epsilon = 0.000001f;
```
Comment (BagelOrb) [inline tests/utils/AABB3DTest.h]: epsilon should be of coord_t rather than floating point.
You are only comparing it to values of coord_t...

Comment (jackha) [inline tests/utils/AABB3DTest.h]: Can you show me how? If I change it some assertions fail and I do not understand why...

Comment (Ghostkeeper) [inline tests/utils/AABB3DTest.h]: `coord_t` is an integer-type so an epsilon of 0.000001 wouldn't make sense. It just gets rounded down to 0. If it's changed to `coord_t` it has a different function.

In this particular case, `epsilon` is being used both to compare the values of transformation matrices (in which case 0.000001f makes sense) and to compare the values of (integer) coordinates in microns (in which case an epsilon of 10 or 15 makes more sense).

**Comment by BagelOrb** [comment]
Made a JRIA ticket: CURA-3830

**Comment by ChrisTerBeke** [comment]
This was closed in Jira because of no activity. It also has conflicts. If we really still want this we can re-open.

**Pull request closed without merging**